src/lua/mod.rs

```diff
@@ -0,0 +1,283 @@
+    }
+}
+
+fn thread_handle_message(sender: &Sender<LuaResponse>, request: LuaQuery, lua: &mut Lua) {
+    match request {
+        LuaQuery::Terminate => {
+            trace!("thread: Received terminate signal");
+            *RUNNING.write().unwrap() = false;
+
+            info!("thread: Lua thread terminating!");
+            return;
+        },
+
+        LuaQuery::Restart => {
+            trace!("thread: Received restart signal!");
+            error!("thread: Lua thread restart not supported!");
+
+            *RUNNING.write().unwrap() = false;
+
+            panic!("Lua thread: Restart not supported!");
+        },
+
+        LuaQuery::Execute(code) => {
+            trace!("thread: Received request to execute code");
+            trace!("thread: Executing {:?}", code);
+
+            match lua.execute::<()>(&code) {
+                Err(error) => {
+                    warn!("thread: Error executing code: {:?}", error);
+                    let response = LuaResponse::Error(error);
+
+                    thread_send(&sender, response);
+                }
+                Ok(_) => {
+                    // This is gonna be really spammy one day
+                    trace!("thread: Code executed okay.");
+                }
+            }
+        },
+
+        LuaQuery::ExecuteFile(name) => {
+            trace!("thread: Received request to execute file {}", name);
+            info!("thread: Executing {}", name);
+
+            let path = Path::new(&name);
+            let try_file = File::open(path);
+
+            if let Ok(file) = try_file {
+                let result = lua.execute_from_reader::<(), File>(file);
+                if let Err(err) = result {
+                    warn!("thread: Error executing {}!", name);
+
+                    thread_send(&sender, LuaResponse::Error(err));
+                }
+                else {
+                    trace!("thread: Execution of {} successful.", name);
+                }
+            }
+            else { // Could not open file
+                // Unwrap_err is used because we're in the else of let
+                let read_error =
+                    LuaError::ReadError(try_file.unwrap_err());
+
+                thread_send(&sender, LuaResponse::Error(read_error));
+            }
+        },
+
+        LuaQuery::GetVariable(varname) => {
+            trace!("thread: Received request to get variable {}", varname);
+            let var_result = lua.get(varname.as_str());
+
+            match var_result {
+                Some(var) => {
+                    thread_send(&sender, LuaResponse::Variable(Some(var)));
+                }
+                None => {
+                    warn!("thread: Unable to get variable {}", varname);
+
+                    thread_send(&sender, LuaResponse::Variable(None));
+                }
+            }
+        },
+
+        LuaQuery::SetValue { name: name, val: val } => {
+        },
+
+        LuaQuery::EmptyArray(name) => {
+            
```

Comment:
If you're not going to fill these in, might want to make it panic with a not implemented (I think there is a macro to work like that, I used it before)